Dockerfile

```diff
@@ -1,361 +1,10 @@
 # Use fixed snapshot of Debian to create a deterministic environment.
 # Snapshot tags can be found at https://hub.docker.com/r/debian/snapshot/tags?name=bullseye
-ARG debian_snapshot=sha256:d647c95797c43e8ef0e0667c4b4acba752ae70385dcf648877903f15c8977da4
-FROM debian/snapshot@${debian_snapshot}
+# ARG debian_snapshot=sha256:d647c95797c43e8ef0e0667c4b4acba752ae70385dcf648877903f15c8977da4
+ARG nix_snapshot=sha256:5073736c16b4c37e49786ef63c4dae7896c9994064ad0873f97c191e3a5bc335
+FROM nixos/nix@${nix_snapshot}
 
-# Set the SHELL option -o pipefail before RUN with a pipe in.
-# See https://github.com/hadolint/hadolint/wiki/DL4006
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN echo 'experimental-features = nix-command flakes' >> /etc/nix/nix.conf
 
-# Uncomment the RUN below if the default snapshot package manager is slow.
```

Comment:
I am new to Nix, so I am not 100% sure how all this fits together. 

Could you point me to where all these dependencies and commands are set up now?

Did you migrate all the info from this file?

E.g., do we still have auto-complete for xtask:

```
# Add sourcing of xtask_bash_completion file to .bashrc		
RUN echo -e "\n#activate xtask auto-complete\nif [ -f /workspace/.xtask_bash_completion ]; then\n  source /workspace/.xtask_bash_completion \nfi" >> "${HOME}/.bashrc"
```

Reply:
I migrated the ones that appear to be used to the `flake.nix` file. I now added some more instructions on how to add new dependencies, but I also already outlined some of those steps in https://github.com/project-oak/oak/blob/main/docs/development.md#quick-usage-guide before.

There are probably some dependencies that are needed for workflows that are not exercised by tests yet (e.g. for makefile-related workflows or other docker images). I can help others add those as we figure them out. Part of the point of the nix config is that we should be re-running those steps too in their own shell, which we currently don't, otherwise we don't even know when they stop working.

If you have some commands that are not covered by tests / CI that you would like to make sure still work, let me know and I can check manually.

Autocomplete for xtask is not included, but it should be feasible, I added a TODO.